src/player.js

```diff
@@ -1963,8 +1963,8 @@ export default class Player extends FakeEventTarget {
       this._eventManager.listen(this, Html5EventType.PAUSE, this._onPause.bind(this));
       this._eventManager.listen(this, Html5EventType.PLAYING, this._onPlaying.bind(this));
       this._eventManager.listen(this, Html5EventType.ENDED, this._onEnded.bind(this));
-      this._eventManager.listen(this, CustomEventType.MUTE_CHANGE, () => {
-        this._playbackAttributesState.muted = this.muted;
+      this._eventManager.listen(this, CustomEventType.MUTE_CHANGE, (event: FakeEvent) => {
+        this._playbackAttributesState.muted = event.payload?.mute || this.muted;
```

Comment:
need to check if there wasn't some usecase that counted on the previous behavior

Reply:
I tested manually the mute/unmute functionality for playlist and single media- seems fine to me. can't think of a use-case where we'd want to ignore the payload if it exists.